subsys/testsuite/ztest/src/ztest.c

```diff
@@ -30,9 +30,11 @@ static bool failed_expectation;
 #include <stdlib.h>
 #include <time.h>
 #include <zephyr/random/random.h>
+#endif
 
-#define NUM_ITER_PER_SUITE CONFIG_ZTEST_SHUFFLE_SUITE_REPEAT_COUNT
-#define NUM_ITER_PER_TEST  CONFIG_ZTEST_SHUFFLE_TEST_REPEAT_COUNT
+#ifdef CONFIG_ZTEST_REPEAT
+#define NUM_ITER_PER_SUITE CONFIG_ZTEST_SUITE_REPEAT_COUNT
+#define NUM_ITER_PER_TEST  CONFIG_ZTEST_TEST_REPEAT_COUNT
```

Comment:
while `CONFIG_ZTEST_SHUFFLE_*_REPEAT` are  deprecated

```suggestion

#define NUM_ITER_PER_SUITE CONFIG_ZTEST_SHUFFLE_SUITE_REPEAT_COUNT
#define NUM_ITER_PER_TEST  CONFIG_ZTEST_SHUFFLE_TEST_REPEAT_COUNT

#ifdef CONFIG_ZTEST_REPEAT
#define NUM_ITER_PER_SUITE CONFIG_ZTEST_SUITE_REPEAT_COUNT
#define NUM_ITER_PER_TEST  CONFIG_ZTEST_TEST_REPEAT_COUNT
```

Reply:
Although it is deprecated it should be existing in the source code?

Reply:
> Although it is deprecated it should be existing in the source code?

yes
https://en.wikipedia.org/wiki/Deprecation#Software

Reply:
Got it. Let me provide some changes.